pkg/test/echo/server/forwarder/config.go

```diff
@@ -158,9 +208,79 @@ func newTLSConfig(r *proto.ForwardEchoRequest, getClientCertificate func(info *t
 		}
 		tlsConfig.RootCAs = certPool
 	}
+
+	// Per-protocol setup.
+	switch c.scheme {
+	case scheme.HTTP, scheme.HTTPS:
+		switch {
+		case r.Http3:
+			if c.scheme == scheme.HTTP {
+				return nil, fmt.Errorf("http3 requires HTTPS")
+			}
+		case r.Http2:
+			if r.Alpn == nil {
+				tlsConfig.NextProtos = []string{"h2"}
+			}
+		default:
+			if r.Alpn == nil {
+				tlsConfig.NextProtos = []string{"http/1.1"}
```

Comment:
AFAIK http/1.1 is the default and doesn't use NextProtos ( unless a new RFC added it - but even then it shouldn't be required).

Reply:
Ack. This is just moved code. Not sure the original motivation here.